src/constructors/mycielskian.c

```diff
@@ -0,0 +1,119 @@
+/*
+   IGraph library.
+   Copyright (C) 2022  The igraph development team <igraph@igraph.org>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "igraph_constructors.h"
+#include "igraph_conversion.h"
+
+#include "igraph_interface.h"
+
+#include "core/interruption.h"
+#include "math/safe_intop.h"
+
+igraph_error_t igraph_mycielskian(igraph_t *res, const igraph_t *graph, igraph_integer_t k) {
+    if (k < 0) {
+        IGRAPH_ERROR("Number of iterations (k) must be non-negative", IGRAPH_EINVAL);
```

Comment:
All error messages must be punctuated. When writing messages, keep in mind that the error message will be shown by different high-level interfaces, within different host languages. Let's try to avoid it looking confusing in some of these contexts, and remove the mention of `k`. 

```suggestion
        IGRAPH_ERROR("The number of Mycielski iterations must not be negative.", IGRAPH_EINVAL);
```